src/transformers/models/mbart50/tokenization_mbart50.py

```diff
@@ -246,7 +246,7 @@ def convert_tokens_to_string(self, tokens):
             else:
                 current_sub_tokens.append(token)
                 prev_is_special = False
-        out_string += self.sp_model.decode(current_sub_tokens)
+        out_string += " " + self.sp_model.decode(current_sub_tokens)
```

Comment:
I am not 100% convinced that this is the fix! 
The reason is that if you always add this, then normal decoding will always add this, which is not expected. 
This should only be added if the previous token is special in the decoding functions. Make sure to add a test based on the issue 🤗

Reply:
Please now check if it's ok or not.